op-service/txintent/txintent_test.go

```diff
@@ -0,0 +1,82 @@
+package txintent
+
+import (
+	"context"
+	"math/big"
+	"math/rand"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/testutils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+)
+
+type call struct {
+	to         *common.Address
+	data       []byte
+	accessList types.AccessList
+}
+
+func (c *call) To() (*common.Address, error)          { return c.to, nil }
+func (c *call) Data() ([]byte, error)                 { return c.data, nil }
+func (c *call) AccessList() (types.AccessList, error) { return c.accessList, nil }
+
+type result struct {
+	blockHash   common.Hash
+	blockNumber uint64
+}
+
+func (r *result) FromReceipt(ctx context.Context, rec *types.Receipt, includedIn eth.BlockRef, chainID eth.ChainID) error {
+	r.blockHash = rec.BlockHash
+	r.blockNumber = includedIn.Number
+	return nil
+}
+func (r *result) Init() Result { return &result{} }
+
+func TestTxIntent(t *testing.T) {
```

Comment:
Maybe you can add a multi-call to this case, or as separate test-case?

Reply:
Added at pull/14948/commits/28bb96558ab3aa6733c3b4c5f460dddd7c7e9d07